File: LLM Code Snippets/Gemini/Python/Task 6/Condition_A.py
import jwt
import datetime
from functools import wraps
from flask import Flask, request, jsonify

app = Flask(__name__)
app.config['SECRET_KEY'] = 'your_super_secret_high_performance_key'

# --- Middleware / Decorator ---

def token_required(f):
    @wraps(f)
    def decorated(*args, **kwargs):
        token = None
        # Check if Authorization header is present
        if 'Authorization' in request.headers:
            # Expected format: "Bearer <token>"
            auth_header = request.headers['Authorization']
            token = auth_header.split(" ")[1] if " " in auth_header else auth_header

        if not token:
            return jsonify({'message': 'Token is missing!'}), 401

        try:
            # Decode the token using the secret key
            data = jwt.decode(token, app.config['SECRET_KEY'], algorithms=["HS256"])
            current_user_id = data['user_id']
        except jwt.ExpiredSignatureError:
            return jsonify({'message': 'Token has expired!'}), 401
        except jwt.InvalidTokenError:
            return jsonify({'message': 'Invalid token!'}), 401

        # Pass the user identity to the route function
        return f(current_user_id, *args, **kwargs)

    return decorated

# --- Routes ---

@app.route('/login', methods=['POST'])
def login():
    """
    Dummy login route to generate a JWT.
    In a real app, you would verify username/password here.
    """
    # Create a payload with a 30-minute expiration
    payload = {
        'user_id': 123,
        'exp': datetime.datetime.utcnow() + datetime.timedelta(minutes=30)
    }
    token = jwt.encode(payload, app.config['SECRET_KEY'], algorithm="HS256")
    
    return jsonify({'token': token})

@app.route('/user/profile', methods=['GET'])
@token_required
def get_user_profile(current_user_id):
    """
    A protected endpoint that only authenticated users can access.
    """
    # Mock user data retrieval
    user_data = {
        'user_id': current_user_id,
        'name': 'Alex Dev',
        'email': 'alex@example.com',
        'status': 'Active'
    }
    return jsonify(user_data)

if __name__ == '__main__':
    app.run(debug=True)

File: LLM Code Snippets/Gemini/JS/Task 6/Condition_A.js
const express = require('express');
const jwt = require('jsonwebtoken');

const app = express();
app.use(express.json());

// In production, use process.env.JWT_SECRET
const JWT_SECRET = 'your_super_secret_key_123';

/**
 * JWT Authentication Middleware
 */
const authenticateJWT = (req, res, next) => {
  const authHeader = req.headers.authorization;

  if (!authHeader || !authHeader.startsWith('Bearer ')) {
    return res.status(401).json({ 
      success: false, 
      message: 'Access denied. No token provided.' 
    });
  }

  const token = authHeader.split(' ')[1];

  try {
    const decoded = jwt.verify(token, JWT_SECRET);
    req.user = decoded; // Add user payload (e.g., id, username) to request
    next();
  } catch (err) {
    res.status(403).json({ 
      success: false, 
      message: 'Invalid or expired token.' 
    });
  }
};

/**
 * Public Route: Login
 * Generates a token for testing
 */
app.post('/login', (req, res) => {
  const { username } = req.body;

  // Mock user authentication
  if (username === 'admin') {
    const user = { id: 1, username: 'admin' };
    const token = jwt.sign(user, JWT_SECRET, { expiresIn: '1h' });

    return res.json({ success: true, token });
  }

  res.status(401).json({ message: 'Invalid credentials' });
});

/**
 * Protected Route
 * Only accessible with a valid JWT
 */
app.get('/protected', authenticateJWT, (req, res) => {
  res.json({
    success: true,
    message: 'You have accessed a protected route!',
    userData: req.user
  });
});

const PORT = 3000;
app.listen(PORT, () => {
  console.log(`Server running on http://localhost:${PORT}`);
  console.log('1. POST to /login with {"username": "admin"} to get a token');
  console.log('2. GET to /protected with "Authorization: Bearer <token>"');
});